nltk/tokenize/punkt.py

```diff
@@ -1062,25 +1062,32 @@ def _col_log_likelihood(count_a, count_b, count_ab, N):
 
         p = count_b / N
         p1 = count_ab / count_a
-        p2 = (count_b - count_ab) / (N - count_a)
 
-        summand1 = (count_ab * math.log(p) +
-                    (count_a - count_ab) * math.log(1.0 - p))
+        p2_denom = N - count_a
+        if p2_denom == 0:
+            p2 = 1
+        else:
+            p2 = (count_b - count_ab) / p2_denom
+
+        print (p, p1, p2, N, count_a, count_b, count_ab)
+
+        summand1 = (count_ab * math.log(p or 1) +
+                    (count_a - count_ab) * math.log(1.0 - p) if p < 1 else 0)
 
-        summand2 = ((count_b - count_ab) * math.log(p) +
-                    (N - count_a - count_b + count_ab) * math.log(1.0 - p))
+        summand2 = ((count_b - count_ab) * math.log(p or 1) +
+                    (N - count_a - count_b + count_ab) * math.log(1.0 - p) if p < 1 else 0)
 
         if count_a == count_ab:
             summand3 = 0
         else:
-            summand3 = (count_ab * math.log(p1) +
-                        (count_a - count_ab) * math.log(1.0 - p1))
+            summand3 = (count_ab * math.log(p1 or 0) +
```

Comment:
> Good work! I believe it should be math.log(p1 or 1) because we want to avoid taking the log of 0.

Ah whoops! Good catch. 👍 

> Btw, do we have tests for this module?

None testing the `log(0)` that I'm aware of. I hadn't written any when I submitted my original  PR. I'm not really sure how this would come up.